normalized-cache-incubating/src/commonMain/kotlin/com/apollographql/cache/normalized/internal/DefaultApolloStore.kt

```diff
@@ -106,10 +106,12 @@ internal class DefaultApolloStore(
   override fun <D : Operation.Data> normalize(
       operation: Operation<D>,
       data: D,
+      errors: List<Error>?,
       customScalarAdapters: CustomScalarAdapters,
   ): Map<String, Record> {
     return operation.normalize(
         data = data,
+        errors = errors,
```

Comment:
Instead of adding an `errors` parameter all the way down, you could merge the errors in a single tree (using `withErrors()`) earlier and just pass a single `dataAndError`. I think it makes sense to have both `writeOperation(data, error)` and `writeOperation(dataAndErrors)` in `ApolloStore` but anything lower that should probably only work with a single tree.

Reply:
What would be the type of `tree` here?

Reply:
Probably a type alias of `Any?` being able to represent all the built in types + Error + Lists + Arrays.

Reply:
Gotcha 👍. Let's see.

Reply:
In 47da415d99743f432393df14e0810b84e663b9b9 I've done that + a few (hopefully) simplifications. 
- Introduce `DataWithErrors` typealias
- remove `readDataFromCache` extensions (use `ApolloStore` instead)
- `normalize` extension still exists, but on the data and renamed `.normalized()`
Sorry this PR has gotten a bit large though 😬